students/Isabella_Kemp/lesson03/mailroom.py

```diff
@@ -0,0 +1,147 @@
+# User can enter a new donor, new donation, user and donation is added to list
+# We finish with a thank you
+def thank_you():
+    while True:
+        name = get_user_input("Enter a donor name to send a thank you letter, "
+                              "'list' or 'listall' for list of donors, '3' will exit: ")
+        if name == "list":
+            for record in donors:
+                print(record[0])  # name
+        elif name == "listall":
+            for record in donors:
+                print(record)  # entire record, name with donations
+        elif name == '3':  # allows them to quit and go back to menu
+            print("Finished processing thanks to donors...\n\n")
+            return
+        else:
+            # User entered a name, so process that
+            record = get_donor(name)
+            #print(record) # for debug comment out
+            if (record != None):
+                # donor is in our list,
+                # get donation amount, add donation to list, send thank you
+                print("user is in our list")  # for debug
+                donation = get_donation()
+                #print(donation)  # for debug
+                record[1].append(donation)
+                #print(record)  # for debug
+                email(name, donation)
+
+            else:
+                # new donor is not in our list
+                # add the donor to our list
+                # get donation amount, add donation to list, send thank you
+                print("!!! NEW DONOR !!!")
+                donation = get_donation()
+                #print(donation)  # for debug
+                new_record = (name, [donation])
+                #print(new_record)  # for debug
+                donors.append(new_record)
+                email(name, donation)
+#end thank_you
+
+
+# checks to see if provided name is in our donor list
+# returns donor record if found, else None
+def get_donor(donor_name):
+    for record in donors:
+        if (record[0] == donor_name):
+            return record  # record[0] is name, record[1] is tuple with variable length of donations
+    return None
+
+# get donation amount
+def get_donation():
+    money = get_user_input("Please enter a donation amount: $ ")
+    amount = float(money)
+    #donors.append(amount)
+    return amount
+
+
+# send thank you to donor
+def email(name, amount):
+    print("Thank you {}, for your generous donation of ${:.2f} !".format(name,amount))
+    return
+
+
+# creates a report of donor name, total donated, number of donations, and average
+# donation amount in each row
+
+def report():
+    print("\n{:<19}| {:<13} | {:<13} | {:>13}".format('Donor Name', 'Total Donated',
+                                                'Number of Donations',
+                                                'Avg. Donation Amount'))
+    print("-"*80)
+    # Takes donors and sorts it representing each item in the list
+    sort_donor = sorted(donors, key = sort_key,reverse = True)
```

Comment:
Spacing off on this section (linter will catch this):
sort_donor = sorted(donors, key=sort_key, reverse=True)